### DISCUSSION TITLE
Example of `@Requires(configuration=String)`?

### DISCUSSION THREAD

**Discussion by jchambers** [micronaut-projects/micronaut-core#7248] [Q&A] [answered]
Hello!

I've successfully used `@Requires` in a number of real-world use cases, but I'm a little fuzzy on using `@Requires(configuration = String)` and am hoping for a little help.

The [docs](https://docs.micronaut.io/latest/api/io/micronaut/context/annotation/Requires.html#configuration--) say:

> Expresses the configurations that should be present for the bean or configuration to load.

…but I'm not exactly sure what "configurations" means in this context.

I have a stanza like this in my `application.yml` file:

```yaml
ejectorSeat:
  enabled: true
```

…and then:

```java
@Context
@Requires(configuration = "ejectorSeat")
public class EjectorSeatSubsystem {
  // ...
}
```

…but that doesn't do quite what I'd expect; the requirement fails and the `EjectorSeatSubsystem` never gets instantiated. Is the `configuration` in this case supposed to be the FQCN of a class annotated with `@ConfigurationProperties`? That's not my read since other `@Requires` properties that refer to classes do, indeed, take a `Class` as an argument. If that's not it, what am I missing?

Thanks kindly!

**Comment by alvarosanchez** [comment, marked as answer]
Requiring a configuration refers to a [bean configuration](https://docs.micronaut.io/latest/guide/#beanConfigurations). 

What you need is the following:

```java
@Context
@Requires(property = "ejectorSeat.enabled", value = "true")
public class EjectorSeatSubsystem {
  // ...
}
```

**Reply by jchambers** [reply]
Ah—thank you. I had missed the whole concept of a bean configuration, and that was, indeed, the missing piece!
